### ISSUE TITLE
Finetuning did not seem to change the generated contents of GPT-2?

### ISSUE THREAD

**Issue by nexthybrid** [karpathy/nanoGPT#161]
Somehow, after my finetuning GPT-2 using the shakespeare dataset, the output result was exactly the same as the stock GPT-2. Did I do something wrong? My commands:

```
cd data/shakespeare
python prepare.py
cd ../..
```

Then I modified line 41 of train.py from `init_from = 'resume'` to `init_from = 'gpt2'`. And then I ran:
python train.py config/finetune_shakespeare.py

On my M2 Macbook Air this took ~ 2 hours to finish training. Then I typed:
```
python sample.py --out_dir=out-shakespeare
```

And I got the following generated contents (the same as what I got from the stock GPT-2):
```
John McAfee, who recently retired from Google's Mountain View offices and now runs C&C, said that while he would like to invest "in this space, I'm afraid I'll sell it before we even sell it" to the public.
...
```

I checked the size of the stock GPT-2 124M model, which is 497.8M. And the fine-tuned model has a size of 1.54GB.

Does anyone know why the finetuning did not seem to change the GPT-2 stock model?

**Comment by venzen** [comment]
Not sure that this is going to solve your issue, but I'd recommend you try the process of finetuning again with the following practice:

1. Execute commands from the project root directory (_nanoGPT/_). Avoid changing directory into subfolders
2. No need to make changes in the core files (such as _nanoGPT/train.py_) because params can be over-ridden by passing them from the command line (e.g. `--init_from=`) or they are over-ridden by the values set in the scripts in _nanoGPT/config/_

I reckon if you reset _train.py_ (and _sample.py_) and follow the walkthrough in the README file it should work for you. :smiley:

**Comment by nexthybrid** [comment]
> Not sure that this is going to solve your issue, but I'd recommend you try the process of finetuning again with the following practice:
> 
> 1. Execute commands from the project root directory (_nanoGPT/_). Avoid changing directory into subfolders
> 2. No need to make changes in the core files (such as _nanoGPT/train.py_) because params can be over-ridden by passing them from the command line (e.g. `--init_from=`) or they are over-ridden by the values set in the scripts in _nanoGPT/config/_
> 
> I reckon if you reset _train.py_ (and _sample.py_) and follow the walkthrough in the README file it should work for you. 😃

Thank you for pointing out that config files overwrite the train.py settings. I did make the `init_from = gpt2` change in the config files, and trained again. However, the loss function value stays stubborn around 2.8 to 3.5 and oscillates. Did Andrej mention changing the learning rate anywhere? I am sticking to what it was in the finetune_skakespeare.py file, which is:

```
learning_rate = 3e-5
decay_lr = False
```

**Comment by harshbhatia66** [comment]
Hi, did this ever get resolved btw?
Came across the same problem where the fine tuned gpt2 model was behaving the same as the original.

**Comment by VatsaDev** [comment]
I finetuned the gpt2 model from the walk through on the readme, and have forked and ran hundreds of times, definitely works
